packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -644,18 +645,30 @@ class DragTarget<T extends Object> extends StatefulWidget {
   _DragTargetState<T> createState() => _DragTargetState<T>();
 }
 
-List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<T>> avatars) {
-  return avatars.map<T?>((_DragAvatar<T> avatar) => avatar.data).toList();
+List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<Object>> avatars) {
+  return avatars.map<T?>((_DragAvatar<Object> avatar) => avatar.data as T?).toList();
 }
 
 class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
-  final List<_DragAvatar<T>> _candidateAvatars = <_DragAvatar<T>>[];
+  final List<_DragAvatar<Object>> _candidateAvatars = <_DragAvatar<Object>>[];
   final List<_DragAvatar<Object>> _rejectedAvatars = <_DragAvatar<Object>>[];
 
+  bool checkDataType(Object? data, Type type) {
```

Comment:
Add documentation here on what `checkDataType` accomplishes. I would note that:
1. On non-web platforms, checks if `data` Object is equal to `type`.
2. On web, it does the same, but requires a check for ints and doubles because `runtimeType` on the web since numbers are treated as doubles on web.

Reply:
I would also describe why this logic is necessary, since it might not be obvious without the context of the issue that made this PR necessary.

Reply:
Might be a more appropriate name for the method:
```suggestion
  bool isExpectedDataType(Object? data, Type type) {
```